api/v3/Activity.php

```diff
@@ -372,6 +407,13 @@ function civicrm_api3_activity_get($params) {
       }
     }
   }
+
+  // Ensure there's enough data for calculating is_overdue
+  if (!empty($options['return']['is_overdue']) && (empty($options['return']['status_id']) || empty($options['return']['activity_date_time']))) {
+    $options['return']['status_id'] = $options['return']['activity_date_time'] = 1;
```

Comment:
Yes. It can't do the calculation without that data.
I suppose we could remove them from the payload afterwards, but I don't think that's super important.